include/morphio/vascProperties.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include <morphio/types.h>
+
+namespace morphio
+{
+namespace VasculatureProperty
+{
+template <typename T>
+void _appendVector(std::vector<T>& to, const std::vector<T>& from, int offset);
+
+struct VascSection
+{
+    using Type = int;
+};
+
+struct Point
+{
+    using Type = morphio::Point;
+};
+
+struct SectionType
+{
+    using Type = morphio::SectionType;
+};
+
+struct Diameter
+{
+    using Type = float;
+};
+
+struct Connection
+{
+    using Type = std::array<int, 2>;
+};
+
+struct VascPointLevel
+{
+    std::vector<Point::Type> _points;
+    std::vector<Diameter::Type> _diameters;
+
+    VascPointLevel() {}
+    VascPointLevel(std::vector<Point::Type> points,
+               std::vector<Diameter::Type> diameters);
+    VascPointLevel(const VascPointLevel& data);
+    VascPointLevel(const VascPointLevel& data, SectionRange range);
+};
+
+struct VascEdgeLevel
+{
+    std::vector<float> leakiness;
+};
+
+struct VascSectionLevel
+{
+    std::vector<VascSection::Type> _sections;
+    std::vector<SectionType::Type> _sectionTypes;
+    std::map<uint32_t, std::vector<uint32_t>> _predecessors;
+    std::map<uint32_t, std::vector<uint32_t>> _successors;
+    std::map<uint32_t, std::vector<uint32_t>> _neighbors;
```

Comment:
The idea is for ease of use right ? Some one who doesn't want to be bothered about directionality would use neighbors. Actually, I was not sure about if successors and neighbors even need to be exposed right now, since we decided not to make the graph directed for now. In that case, I would make them private and only let the user use neighbors.
@eleftherioszisis do you have any comments ?

Reply:
I haven't said we don't need the public function. I am just talking about the data structure itself.